code/game/objects/items.dm

```diff
@@ -546,6 +519,23 @@ var/global/image/fire_overlay = image("icon" = 'icons/goonstation/effects/fire.d
 			throw_at(S,14,3)
 		else ..()
 
+/obj/item/throw_impact(atom/A)
+	if(A && !qdeleted(A))
+		var/itempush = 1
+		if(w_class < 4)
+			itempush = 0 // too light to push anything
+		return A.hitby(src, 0, itempush)
+
+/obj/item/throw_at(atom/target, range, speed, mob/thrower, spin=1, diagonals_first = 0, datum/callback/callback)
+	thrownby = thrower
+	callback = CALLBACK(src, .proc/after_throw, callback) //replace their callback with our own
+	. = ..(target, range, speed, thrower, spin, diagonals_first, callback)
+
+/obj/item/proc/after_throw(datum/callback/callback)
+	if(callback) //call the original callback
+		. = callback.Invoke()
+	throw_speed = initial(throw_speed) //explosions change this.

```

Comment:
Yeah they still don't change it though, I didn't want to touch the explosion system